hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
Ah, I assumed you were using it to copy to flash. If you're not using it to copy to flash, it doesn't need to be RAM-resident, does it?

Your only actual flash operations are "erase" and "program from already-loaded data".

So actually, the only stuff really running during the flash operation that need to be RAM resident are "executeCommand", "launchCommand" and "controllerCurrentlyBusy". No?

(With launchCommand and controllerCurrentlyBusy being simple enough to be inline or macros).